### PR TITLE
chore(curriculum): use ast helpers step 5 case converter

### LINKED ISSUE #56836: [Learn Python List Comprehension by Building a Case Converter Program - Step 5]   f'string interpolation fails lesson but concatenation passes
### Describe the Issue

# [Learn Python List Comprehension by Building a Case Converter Program - Step 5]   f'string interpolation fails lesson but concatenation passes

- Only reproduced this issue in Chorme.130
    

### Affected Page

https://www.freecodecamp.org/learn/scientific-computing-with-python/learn-list-comprehension-by-building-a-case-converter-program/step-5

### Your code

```python
if char.isupper():
            
            converted_character = f'_{char.lower()}'     # fails test
            converted_character = '_' + char.lower()    # passes test  


# Sorry, your code does not pass. Keep trying.

# You should assign the modified character to a variable named converted_character.![Image](https://github.com/user-attachments/assets/40669693-8e7b-4dc4-b2cf-3a4e499286d0)




```

### Expected behavior

f'string should pass test

### Screenshots

_No response_

### System

- Device: Laptop
- OS:  Windows 10
- Browser:Chrome
- Version: 130

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by a2937**
Checklist:

- [X] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [X] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [X] My pull request targets the `main` branch of freeCodeCamp.
- [X] I have tested these changes either locally on my machine, or GitPod.

Closes #56836

It's a little ugly but both tests pass.

### CODE REVIEW

File: curriculum/challenges/english/07-scientific-computing-with-python/learn-list-comprehension-by-building-a-case-converter-program/657efa642593c5746acc5c81.md [hunk 1]
```diff
@@ -18,11 +18,7 @@ You should assign the modified character to a variable named `converted_characte
 ```js
 ({
     test: () => {
-        const transformedCode = code.replace(/\r/g, "");
-        const convert_to_snake_case = __helpers.python.getDef("\n" + transformedCode, "convert_to_snake_case");
-        const { function_body } = convert_to_snake_case;
-
-        assert.match(function_body, / +converted_character\s*=\s*("|')_\1\s*\+\s*char\.lower\s*\(\s*\)/);
+     assert(runPython(`_Node(_code).find_function("convert_to_snake_case").find_for_loops()[0].find_body().find_ifs()[0].find_bodies()[0].find_variable('converted_character').is_equivalent("converted_character = '_' + char.lower()") or _Node(_code).find_function("convert_to_snake_case").find_for_loops()[0].find_body().find_ifs()[0].find_bodies()[0].find_variable('converted_character').is_equivalent("converted_character = f'_{char.lower()}'")`));
```
Comment (gikf) [inline curriculum/challenges/english/07-scientific-computing-with-python/learn-list-comprehension-by-building-a-case-converter-program/657efa642593c5746acc5c81.md:21]:
```suggestion
assert(runPython(`
  converted_character_variable = _Node(_code).find_function("convert_to_snake_case").find_for_loops()[0].find_ifs()[0].find_variable('converted_character')
  
  converted_character_variable.is_equivalent("converted_character = '_' + char.lower()") or converted_character_variable.is_equivalent("converted_character = f'_{char.lower()}'")
`));
```
Trying to make it a bit simpler.


I've also thrown it into python formatter (https://black.vercel.app/) 👀 :
```suggestion
    assert(runPython(`
      converted_character_variable = (
          _Node(_code)
          .find_function("convert_to_snake_case")
          .find_for_loops()[0]
          .find_ifs()[0]
          .find_variable("converted_character")
      )
      
      converted_character_variable.is_equivalent(
          "converted_character = '_' + char.lower()"
      ) or converted_character_variable.is_equivalent(
          "converted_character = f'_{char.lower()}'"
      )
    `)); 
```

File: curriculum/challenges/english/07-scientific-computing-with-python/learn-list-comprehension-by-building-a-case-converter-program/657efa642593c5746acc5c81.md [hunk 2]
```diff
@@ -32,9 +28,7 @@ You should not have `pass` in your `if` statement.
 ```js
 ({
     test: () => {
-        const commentless_code = __helpers.python.removeComments(code);
-        const {block_body} = __helpers.python.getBlock(commentless_code, /if\s+char\.isupper\s*\(\s*\)\s*/);
-        assert.notMatch(block_body, /pass/);
+        assert(runPython(`_Node(_code).find_function("convert_to_snake_case").find_for_loops()[0].find_body().find_ifs()[0].find_bodies()[0].has_stmt("pass") == False`));
```
Comment (gikf) [inline curriculum/challenges/english/07-scientific-computing-with-python/learn-list-comprehension-by-building-a-case-converter-program/657efa642593c5746acc5c81.md:31]:
```suggestion
        assert(runPython(`not _Node(_code).find_function("convert_to_snake_case").find_for_loops()[0].find_ifs()[0].has_stmt("pass")`)); 
```

**Review by gikf — COMMENTED** [review]
Just a couple nitpicks

File: curriculum/i18n-curriculum
[the remark this answers is not in the document]
Comment (a2937) [reply curriculum/i18n-curriculum:1]: Dang it. Why does this keep getting grabbed?

**Review by gikf — APPROVED** [review]

**Review by majestic-owl448 — APPROVED** [review]

**Pull request merged**